### ISSUE TITLE
[FIXED] Red LED always on and tCam will not start up after charging. [SOLUTION] Remove and re-insert battery connector.

### ISSUE THREAD

**Issue by dudeitssm** [danjulio/tCam#13]
### TLDR
The EFM8SB20F32G can lock up on older firmware versions. 
This results in the tCam being unable to boot, and the red LED being lit.

### Immediate fix
Disconnecting and reconnecting battery connector results in rebooting the processor, and the tCam can start up.

### Long term solution
Update old PMIC firmware to latest following the [instructions](https://github.com/danjulio/gCore/tree/main/Firmware) here.

-----

Hi Dan.

I was using my tCam normally. Once the battery ran low, I shut down the tCam. At this point, the LED was off.
When I plugged it into my USB-C charger, the red LED came on, and I went on with my day.
This is the charger I have used many times before without issues: a [DeWalt 65W USB-C charger](https://www.homedepot.com/p/DEWALT-20V-MAX-Charger-and-USBC-Adaptor-Kit-DCB094K/318666174).

After a couple hours, I checked up on it and the red LED was still on.
I unplugged it from the charger and... the red LED still remained lit.

I pressed the power button and it wouldn't start up.
I pressed and held the power button for over 1 minute and it still would not start up, and the red LED remains lit.

Looking through the [gcore schematic](https://github.com/danjulio/gCore/blob/main/Documentation/Documents/gCore_schem.pdf), it seems the LEDs are controlled by the EFM8SB20F32G. I could be mistaken though -- my second guess would be the ESP32 has died.

I would open up the tcam and probe it for troubleshooting, but unfortunately, I do not have the tiny allen key required to open it. I will order a set online so I can in a few days.

Any pointers would be appreciated in the mean time.
Thank you!

**Comment by danjulio** [comment]
Probably what happened was that the EFM8 crashed or hung-up.  So, unfortunately, you will have to disassemble it to disconnect and then reconnect the battery to reset it.  You can leave it plugged in to the charger to prevent the battery from being over-discharged as the actual charger is a separate IC.

When it’s up and running check to see what version of firmware the EFM8 is running (under the Settings/Info button you’ll see RTC/PMIC version).  You might update the firmware if it is running version 1.1 (there are instructions how at the link below).

https://github.com/danjulio/gCore/tree/main/Firmware <https://github.com/danjulio/gCore/tree/main/Firmware>

Let me know if you have any other questions.

Regards, Dan

**Comment by dudeitssm** [comment]
Thank you, Dan.

I will let you know once I have the allen keys to disassemble and disconnect/reconnect the battery.

**Comment by dudeitssm** [comment]
Yup, disconnecting and reconnecting the battery revived the tcam. That's a relief.

In case anyone is curious, here's the version info:

```
FW Version: 0.20
SDK Version: v4.4.4-268-ge47aaa7088
tCam-Mini Version: 3.1
RTC/PMIC Version: 1.1
```

**Comment by danjulio** [comment]
I’m glad that fixed it for you!

I have a question for you if you don’t mind: Was tCam on or off when the EFM8 locked up?

I released version 1.2 of the firmware for the EFM8 (RTC/PMIC). Mainly it fixed some issues that slowed down the RTC but it also fixed a potential lock-up.  However I think that lock-up required the ESP32 to be accessing the EFM8 via the I2C bus.  You can use a dev board like a ESP32, Teensy or RP2040 programmed via the Arduino IDE to load it if you want.  Instructions are in the firmware subdirectory of the repo.

Otherwise there might still be some rare bug lurking in that code.  

Regards, Dan

**Comment by dudeitssm** [comment]
Hi Dan.

> I have a question for you if you don’t mind: Was tCam on or off when the EFM8 locked up?

I turned the tcam off before charging. 

I'll think about flashing the updated PMIC firmware at another time. Especially since this problem seems to appear once in a blue moon (first time in, eh, 2 years or so?).

The [instructions](https://github.com/danjulio/gCore/tree/main/Firmware#local-firmware-update) provided are easy to follow (to those familiar with hardware flashing, which I am).

Thanks again :+1:
